AsyncDisplayKit/Layout/ASLayoutSpec.mm

```diff
@@ -45,7 +44,7 @@ - (instancetype)init
   }
   _isMutable = YES;
   _environmentState = ASEnvironmentStateMakeDefault();
-  
+  _children = [NSArray array];
```

Comment:
@rcancro lolol.  I think this is cheating a bit instead of just checking for nil in the necessary case, which I think is only one place, right?  In that one case we can simply do = @[object], instead of appendObject:, so it should be clean.  

These empty arrays aren't heavy, but we do sometimes create large layout trees on every frame during a gesture like in the commerce scroller resizing, and this would create many dozens of extra objects that can be skipped (not saying it is an influential performance problem, moreso a best practice for a core base class).  Suggestion: if you like this pattern, create a singleton empty array and always assign to that pointer.  _maybe_ NSArray array already does that?  We could check if the pointer is always the same?